api/sip.py

```diff
@@ -0,0 +1,596 @@
+        64<patron status><language><transaction date><hold items count><overdue items count>
+        <charged items count><fine items count><recall items count><unavailable holds count>
+        <institution id><patron identifier><personal name><hold items limit><overdue items limit>
+        <charged items limit><valid patron><valid patron password><currency type><fee amount>
+        <fee limit><items><home address><e-mail address><home phone number><screen message><print line>
+        
+        patron status: 14-char, required
+        language: 3-char, req
+        transaction date: 18-char, YYYYMMDDZZZZHHMMSS, required
+        hold items count: 4-char, required
+        overdue items count: 4-char, required
+        charged items count: 4-char, required
+        fine items count: 4-char, required
+        recall items count: 4-char, required
+        unavailable holds count: 4-char, required
+        institution id: AO, variable-length, required
+        patron identifier: AA, var-length, req
+        personal name: AE, var-length, req
+        hold items limit: BZ, 4-char, optional
+        overdue items limit: CA, 4-char, optional
+        charged items limit: CB, 4-char, optional
+        valid patron: BL, 1-char, Y/N, optional
+        valid patron password: CQ, 1-char, Y/N, optional
+        currency type: BH, 3-char, optional
+        fee amount: BV, var-length.  The amount of fees owed by this patron.
+        fee limit: CC, variable-length, optional
+        items: 0 or more instances of one of the following, based on "summary" field of patron information message
+            hold items: AS, var-length opt (should be sent for each hold item)
+            overdue items: AT, var-length opt (should be sent for each overdue item)
+            charged items: AU, var-length opt (should be sent for each charged item)
+            fine items: AV, var-length opt (should be sent for each fine item)
+            recall items: BU, var-length opt (should be sent for each recall item)
+            unavailable hold items: CD, var-length opt (should be sent for each unavailable hold item)
+            home address: BD, variable-length, optional
+            email address: VE, variable-length, optional
+            home phone number: BF, variable-length optional
+
+        screen message: AF, var-length, optional
+        print line: AG, var-length, optional
+        """
+        return self.parse_response(
+            data,
+            64,
+            fixed.patron_status,
+            fixed.language,
+            fixed.transaction_date,
+            fixed.hold_items_count,
+            fixed.overdue_items_count,
+            fixed.charged_items_count,
+            fixed.fine_items_count,
+            fixed.recall_items_count,
+            fixed.unavailable_holds_count,
+            named.institution_id.required,
+            named.patron_identifier.required,
+            named.personal_name.required,
```

Comment:
Maybe I'm being too clever here. `named.personal_name.required` calls the `named.required` property, which returns a `named` object that is "personal name, except required". So this _is_ the code where the named fields are made required.